File: support_tools/swigwin-3.0.12/Examples/test-suite/go/director_nested_runme.go
package main

import . "./director_nested"

type A struct{} // From FooBar_int
func (p *A) Do_step() string {
	return "A::do_step;"
}
func (p *A) Get_value() string {
	return "A::get_value"
}

func f1() {
	a := NewDirectorFooBar_int(&A{})
	if a.Step() != "Bar::step;Foo::advance;Bar::do_advance;A::do_step;" {
		panic("Bad A virtual resolution")
	}
}

type B struct{} // From FooBar_int
func (p *B) Do_advance() string {
	return "B::do_advance;" + p.Do_step()
}
func (p *B) Do_step() string {
	return "B::do_step;"
}
func (p *B) Get_value() int {
	return 1
}

func f2() {
	b := NewDirectorFooBar_int(&B{})

	if b.Step() != "Bar::step;Foo::advance;B::do_advance;B::do_step;" {
		panic("Bad B virtual resolution")
	}
}

type C struct {
	fbi FooBar_int
} // From FooBar_int

func (p *C) Do_advance() string {
	return "C::do_advance;" + DirectorFooBar_intDo_advance(p.fbi)
}

func (p *C) Do_step() string {
	return "C::do_step;"
}

func (p *C) Get_value() int {
	return 2
}

func (p *C) Get_name() string {
	return DirectorFooBar_intGet_name(p.fbi) + " hello"
}

func f3() {
	m := &C{nil}
	cc := NewDirectorFooBar_int(m)
	m.fbi = cc
	c := FooBar_intGet_self(cc)
	c.Advance()

	if c.Get_name() != "FooBar::get_name hello" {
		panic(0)
	}

	if c.Name() != "FooBar::get_name hello" {
		panic(0)
	}
}

func main() {
	f1()
	f2()
	f3()
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/python/director_nested_runme.py
from director_nested import *


class A(FooBar_int):

    def do_step(self):
        return "A::do_step;"

    def get_value(self):
        return "A::get_value"

    pass


a = A()
if a.step() != "Bar::step;Foo::advance;Bar::do_advance;A::do_step;":
    raise RuntimeError, "Bad A virtual resolution"


class B(FooBar_int):

    def do_advance(self):
        return "B::do_advance;" + self.do_step()

    def do_step(self):
        return "B::do_step;"

    def get_value(self):
        return 1

    pass


b = B()

if b.step() != "Bar::step;Foo::advance;B::do_advance;B::do_step;":
    raise RuntimeError, "Bad B virtual resolution"


class C(FooBar_int):

    def do_advance(self):
        return "C::do_advance;" + FooBar_int.do_advance(self)

    def do_step(self):
        return "C::do_step;"

    def get_value(self):
        return 2

    def get_name(self):
        return FooBar_int.get_name(self) + " hello"

    pass

cc = C()
c = FooBar_int_get_self(cc)
c.advance()

if c.get_name() != "FooBar::get_name hello":
    raise RuntimeError

if c.name() != "FooBar::get_name hello":
    raise RuntimeError
